controller/secret.go

```diff
@@ -149,18 +159,17 @@ func (sc *SecretController) upsertSecret(saName, saNamespace string) {
 	}
 
 	if exists {
-		// TODO (Yang Guan): https://github.com/istio/auth/issues/56
-		// For an existing secret, we should check the cert expiration time and refresh it when necessary.
+		// Do nothing for existing secrets. Rotating expiring certs are handled by the `scrtUpdated` method.
 		return
 	}
 
 	// Now we know the secret does not exist yet. So we create a new one.
 	chain, key := sc.ca.Generate(saName, saNamespace)
 	rootCert := sc.ca.GetRootCertificate()
 	secret.Data = map[string][]byte{
-		"cert-chain.pem": chain,
-		"key.pem":        key,
-		"root-cert.pem":  rootCert,
+		certChainSecretKey:  chain,
```

Comment:
I was following the convention used in kubernetes code base but I can see the confusion. I guess we can rename it to `rootCertSecreteMapKey` to make it clear that it's a key in a map.